cloud/google/machineactuator.go

```diff
@@ -156,7 +160,7 @@ func (gce *GCEClient) CreateMachineController(cluster *clusterv1.Cluster, initia
 	if gce.machineSetupConfigGetter == nil {
 		return errors.New("a valid machineSetupConfigGetter is required")
 	}
-	if err := gce.CreateMachineControllerServiceAccount(cluster, initialMachines); err != nil {
+	if err := gce.serviceAccountService.CreateMachineControllerServiceAccount(cluster); err != nil {
```

Comment:
Once again, what was the original intent of providing a list of initial nodes during the `CreateMachineController` operation? The signature is part of the `machineDeployer` interface which is not exported but the `GCEClient` client actually implements all its methods. So the interface should be exported as `MachineDeployer` and at least the `var _ MachineDeployer = &GCEClient{}` construct used in the `cloud/google/machineactuator.go` to have the compiler check the client implements the interface.

Reply:
gcpDeployer should be removed soon because it has been replaced by clusterctl (#230), this will remove the machineDeployer interface. 

In general the machines are needed when information from the machine config is needed. In the case of `CreateMachineController` the zone is needed by `setupSSHAccess`. The reason these other methods needed it originally was to get the project field, which has since moved onto the cluster config instead of the machines.